packages/react-test-renderer/src/ReactShallowRenderer.js

```diff
@@ -237,25 +268,34 @@ class ReactShallowRenderer {
         // current hook.
         const queue: UpdateQueue<A> = (workInProgressHook.queue: any);
         const dispatch: Dispatch<A> = (queue.dispatch: any);
+        if (this._nonRenderPhaseUpdates !== null) {
+          const possibleStateAndDispatcher = this._getPreviousStateAndDispatcherFromUpdateMap(
+            workInProgressHook,
+            reducer,
+            this._nonRenderPhaseUpdates,
+            queue,
+            dispatch,
+          );
+          if (possibleStateAndDispatcher !== null) {
+            this._nonRenderPhaseUpdates = null;
+            // A sanity check to ensure two maps do not get used together
+            invariant(
+              this._didScheduleRenderPhaseUpdate === false,
+              'non-render phase updates should never occur in combination with render phase updates',
```

Comment:
Why would they not?

Reply:
I couldn't think of a case where that would happen, at least in the case they do this invariant would hit. If you can think of a test for it I can enable it and add the logic for that code path :)